File: OpenControl/Core/Network/DTO/Consultation/Consultation.swift
//
//  Consultation.swift
//  OpenControl
//
//  Created by Vladislav Panev on 21.05.2023.
//

import Foundation

struct Consultation: Codable {
    let id: String
    let topic: String?
    let starDateTime: Date?
    let finishDateTime: Date?
    let videoRecordPath: String?
    let otherInformation: String?

    let userId: CLong
    let departmentUserId: CLong?
    let consultationSlotId: String
}

File: Back-end/Leader.Entity/Entity/Consultation.cs
namespace Leader.Domain.Entity;

public class Consultation
{
    public Guid Id { get; set; }
    public string? Topic { get; set; }
    public DateTime? StarDateTime { get; set; }
    public DateTime? FinishDateTime { get; set; }
    public string? VideoRecordPath { get; set; }
    public string? OtherInformation { get; set; }

    public long? UserId { get; set; }
    public User? User { get; set; }

    public long? DepartmentUserId { get; set; }
    public DepartmentUser? DepartmentUser { get; set; }

    public Guid? ConsultationSlotId { get; set; }
    public ConsultationSlot? ConsultationSlot { get; set; }
}